lucene/core/src/java/org/apache/lucene/codecs/blocktree/FieldReader.java

```diff
@@ -98,7 +97,7 @@
           isFSTOffHeap = ((this.docCount != this.sumDocFreq) || openedFromWriter == false);
           break;
         case AUTO:
-          isFSTOffHeap = ((this.docCount != this.sumDocFreq) || openedFromWriter == false) && indexIn instanceof ByteBufferIndexInput;
+          isFSTOffHeap = ((this.docCount != this.sumDocFreq) || openedFromWriter == false) &&  indexIn.load(); // NOCOMMIT - should we do this should we do it all the time?
```

Comment:
This looks reasonable to me. The extreme minority of expert users who don't want to preload this slice can still use a Directory wrapper whose index inputs don't do anything in load() and return `true`.

Reply:
+1

Extra space after the `&&`?